app/src/ui/repository.tsx

```diff
@@ -340,3 +351,78 @@ export class RepositoryView extends React.Component<
     )
   }
 }
+
+type KeyStatus = 'new' | 'updated' | 'removed' | 'no change'
```

Comment:
All code related to diffObj will be removed as I only added it to try to track down state changes.